alternator/executor.hh

```diff
@@ -24,6 +24,7 @@
 #include "utils/updateable_value.hh"
 
 #include "tracing/trace_state.hh"
+#include "db/config.hh"
```

Comment:
Please don't add db/config.hh to headers. It amplifies changes - any component can change db/config.hh, and then alternator has to be rebuilt.

Reply:
We're already pretty-much screwed in that respect - even if header files don't include db/config.hh, still ever source file that needs anything from the configuration includes it. A quick grep counted 126 (!) such source files.

The reason why I found myself including it in a **header** file is that I need the type db::tri_mode_restriction in that header file (as a parameter type), and that type is defined in db/config.hh. I can fix this by moving tri_mode_restriction to its own header file, and include it in db/config.hh - and here in alternator/executor.hh. I'll do that.